File: sdks/python/reshadx/resources/credit_score.py
"""ReshADX Python SDK - Credit Score Resource"""

from typing import Any, Dict, List, Optional

from ..models.credit import CalculateCreditScoreResponse, CreditRecommendation, CreditScore
from ..utils.http import HttpClient


class CreditScoreResource:
    """Credit Score resource"""

    def __init__(self, http: HttpClient):
        self.http = http

    def calculate(self, include_alternative_data: bool = False) -> CalculateCreditScoreResponse:
        """Calculate credit score"""
        data = {"includeAlternativeData": include_alternative_data}

        response = self.http.post("/credit-score/calculate", json_data=data)
        return CalculateCreditScoreResponse(**response)

    def get(self) -> CreditScore:
        """Get current credit score"""
        response = self.http.get("/credit-score")
        return CreditScore(**response)

    def get_history(
        self,
        start_date: Optional[str] = None,
        end_date: Optional[str] = None,
        limit: int = 10,
    ) -> Dict[str, List[Dict[str, Any]]]:
        """Get credit score history"""
        params: Dict[str, Any] = {"limit": limit}

        if start_date:
            params["startDate"] = start_date
        if end_date:
            params["endDate"] = end_date

        return self.http.get("/credit-score/history", params=params)

    def get_factors(self) -> Dict[str, List[Dict[str, Any]]]:
        """Get credit score factors"""
        return self.http.get("/credit-score/factors")

    def get_recommendations(self) -> List[CreditRecommendation]:
        """Get credit recommendations"""
        response = self.http.get("/credit-score/recommendations")
        return [CreditRecommendation(**rec) for rec in response["recommendations"]]

    def simulate(
        self,
        pay_off_debt: Optional[int] = None,
        increase_income: Optional[int] = None,
        reduce_utilization: Optional[int] = None,
    ) -> Dict[str, Any]:
        """Get credit score simulator"""
        scenarios = {}

        if pay_off_debt is not None:
            scenarios["payOffDebt"] = pay_off_debt
        if increase_income is not None:
            scenarios["increaseIncome"] = increase_income
        if reduce_utilization is not None:
            scenarios["reduceUtilization"] = reduce_utilization

        return self.http.post("/credit-score/simulate", json_data=scenarios)

File: sdks/typescript/src/resources/credit-score.ts
/**
 * ReshADX TypeScript SDK - Credit Score Resource
 */

import { HttpClient } from '../utils/http';
import {
  CalculateCreditScoreRequest,
  CalculateCreditScoreResponse,
  CreditScore,
  CreditScoreFactor,
  CreditRecommendation,
} from '../types';

export class CreditScoreResource {
  constructor(private http: HttpClient) {}

  /**
   * Calculate credit score
   */
  async calculate(request: CalculateCreditScoreRequest = {}): Promise<CalculateCreditScoreResponse> {
    return this.http.post<CalculateCreditScoreResponse>('/credit-score/calculate', request);
  }

  /**
   * Get current credit score
   */
  async get(): Promise<CreditScore> {
    return this.http.get<CreditScore>('/credit-score');
  }

  /**
   * Get credit score history
   */
  async getHistory(params: {
    startDate?: string;
    endDate?: string;
    limit?: number;
  } = {}): Promise<{
    scores: Array<CreditScore & { calculatedAt: string }>;
  }> {
    return this.http.get('/credit-score/history', { params });
  }

  /**
   * Get credit score factors
   */
  async getFactors(): Promise<{ factors: CreditScoreFactor[] }> {
    return this.http.get('/credit-score/factors');
  }

  /**
   * Get credit recommendations
   */
  async getRecommendations(): Promise<{ recommendations: CreditRecommendation[] }> {
    return this.http.get('/credit-score/recommendations');
  }

  /**
   * Get credit score simulator
   */
  async simulate(scenarios: {
    payOffDebt?: number;
    increaseIncome?: number;
    reduceUtilization?: number;
  }): Promise<{
    currentScore: number;
    projectedScore: number;
    improvement: number;
    timeframe: string;
  }> {
    return this.http.post('/credit-score/simulate', scenarios);
  }
}
